src/main/flight/pid.c

```diff
@@ -990,6 +996,22 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, const rollAndPitchT
             // Only increase ITerm if output is not saturated
             pidData[axis].I = ITermNew;
         }
+        #if defined(USE_TBH)
+            if (pidProfile->iterm_tbh && (axis < FD_YAW || pidProfile->iterm_tbh == ITERM_TBH_RPY)) {
+                // Non-linear Take-Back Half algorithm
+                // When the error term crosses through zero, this algorithm causes
+                // the I term to non-linearly jump to half-way between the current
+                // accumulated error and the accumulated error at the last zero
+                // crossing.
+                const bool thisSign = itermErrorRate > 0; //T==Pos, F==Neg or zero
+                if (thisSign != tbhErrorSignPrev[axis]) { 
```

Comment:
I'd prefer storing `itermErrorRatePrev` rather than boolean for sign and then checking for  `itermErrorRate * itermErrorRatePrev < 0`